snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -441,14 +441,26 @@ def _assemble_runtime_environment(self) -> str:
             # https://bugs.launchpad.net/snapd/+bug/1860369
             return ""
 
-        common.env = self._project_config.snap_env()
-        assembled_env = common.assemble_env()
-        common.reset_env()
-
-        assembled_env = assembled_env.replace(self._prime_dir, "$SNAP")
-        assembled_env = self._install_path_pattern.sub("$SNAP", assembled_env)
-        ld_library_env = "export LD_LIBRARY_PATH=$SNAP_LIBRARY_PATH:$LD_LIBRARY_PATH"
-        return "\n".join([assembled_env, ld_library_env])
+        env = list()
+        if self._project_config.project._snap_meta.base in ("core", "core16", "core18"):
+            common.env = self._project_config.snap_env()
+            assembled_env = common.assemble_env()
+            common.reset_env()
+
+            assembled_env = assembled_env.replace(self._prime_dir, "$SNAP")
+            env.append(self._install_path_pattern.sub("$SNAP", assembled_env))
+        else:
+            # TODO use something local to the meta package and
+            # only add paths for directory items that actually exist.
+            runtime_env = project_loader.runtime_env(
+                self._prime_dir, self._project_config.project.arch_triplet
+            )
+            for e in runtime_env:
+                env.append(re.sub(self._prime_dir, "$SNAP", e))
```

Comment:
Feels like runtime_env() should do that conversion.

Reply:
This instance of runtime_env is used everywhere in a very different way, I import the logic here until we get rid of that.

Reply:
this could probably be done while working on the TODO I added.